src/http-client/fetch-client.ts

```diff
@@ -14,13 +14,23 @@ export class FetchClient implements HTTPClient {
     headers: requestHeaders,
     method,
     url,
+    client_timeout_ms,
   }: HTTPRequest): Promise<HTTPResponse> {
     // TODO: handle client timeouts with AbortController. Emit NetworkError if so.
 
+    let controller: AbortController;
+    let signal: AbortSignal | undefined = undefined;
+    if (client_timeout_ms !== undefined) {
```

Comment:
this concept should never be undefined.

ALso, let's use the buffer here added to the query timeout.